### PR TITLE
Create a `rustc_codegen_c` repository

### PR DESCRIPTION
**Pull request by tgross35**
One of the OSPP projects this year is a C codegen backend. Create a repository where this will be developed, modeled off the codegen_cranelift repo file.

r? @Kobzol since I think you are in the loop with this

### CODE REVIEW

File: repos/rust-lang/rustc_codegen_c.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+org = 'rust-lang'
+name = 'rustc_codegen_c'
+description = 'C based backend for rustc'
+bots = []
+
+[access.teams]
+compiler = 'maintain'
+
+[access.individuals]
+tgross35 = 'maintain'
```
Comment (ehuss) [inline repos/rust-lang/rustc_codegen_c.toml:10]: We have a policy of not giving individual access anymore. Can you find a team that would be appropriate to add here instead?

Comment (tgross35) [reply repos/rust-lang/rustc_codegen_c.toml:10]: Compiler makes sense (and is included). I'm just not a member of any relevant team but am the mentor for this project. Any suggestions?

Maybe I could join e.g. compiler contributors, but I have no clue what is required there.

Comment (ehuss) [reply repos/rust-lang/rustc_codegen_c.toml:10]: The only other suggestion I have is to create a subteam of compiler responsible for this backend. You might want to discuss with the compiler team.

**Comment by tgross35** [comment]
I updated this to be both compiler and libs with contributors, which covers me. I think this is fine, if not then we might just need a new group.

@wesleywiser can you confirm?

**Review by Amanieu — APPROVED** [review]

**Review by wesleywiser — APPROVED** [review]

**Comment by jackh726** [comment]
Not sure that adding libs/libs-contributors here is the right choice. I think the most lightweight option is to make a marker team like "ospp-2024-codegen-c" to add both you and potentially the ospp participant?

**Comment by tgross35** [comment]
Sorry @jackh726, I missed your comment. I changed this to create a subteam `codegen-c-maintainers` (opted against including OSPP-2024 since this project will outlive it) with Amanieu (OSPP organizer) and myself (advisor for this project) as members. This team and `compiler` can maintain, `compiler-contributors` has write.

This is about the same as cg_gcc.

**Review by jackh726 — COMMENTED** [review]
Going to go ahead and merge this. 

Am going to cc @wesleywiser and @davidtwco here - let me know if you have concerns

**Review by jackh726 — APPROVED** [review]

**Pull request merged**